### PR TITLE
Add Category and Arrow instances for unfolds

### PR DESCRIPTION
**Pull request by harendra-kumar**
Just moved the functions around in the lower level module, to add Category and Arrow instances. All the functionality was already there, just used the existing functions in the instances.

### CODE REVIEW

File: src/Streamly/Internal/Data/Unfold.hs [hunk 1]
```diff
@@ -8,6 +8,13 @@
 -- Stability   : experimental
 -- Portability : GHC
 --
+-- The 'Unfold' abstraction is less powerful but more flexible than the Stream
+-- abstraction.  Stream is an unfold with an embedded state. Therefore, Stream
+-- is like an object whereas unfold is like a function.  Unfold is a 'Category'
+-- and an 'Arrow' but not a 'Monad'. Whereas Stream is not a Category or Arrow
```
Comment (adithyaov) [inline src/Streamly/Internal/Data/Unfold.hs:14]: Unfold is a Monad now. Do you want to change the wording?

Comment (harendra-kumar) [reply src/Streamly/Internal/Data/Unfold.hs:14]: Doing that right now.

**Review by adithyaov — COMMENTED** [review]
Did you check if the laws for the instances hold?

**Comment by harendra-kumar** [comment]
> Did you check if the laws for the instances hold?

Did not do a rigorous proof, but I thought about it quickly and it looked ok.

**Review by adithyaov — APPROVED** [review]

**Pull request merged**